### PR TITLE
update ci

### PR DESCRIPTION
**Pull request by Emilgardis**
part of #724

remove all uses of actions-rs which use deprecated gha functions and is not maintained
removes specific clippy-check in favor of running clippy directly

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,12 +10,13 @@ jobs:
     name: CI
     runs-on: ubuntu-latest
     needs: [check, ci-linux, ci-clippy, ci-serde]
+    if: always()
     steps:
       - name: Done
-        run: exit 0
+        run: jq --exit-status 'all(.result == "success")' <<< '${{ toJson(needs) }}'
```
Comment (Emilgardis) [inline .github/workflows/ci.yml:16]: preparation for merge queue

**Review by Emilgardis — COMMENTED** [review]
one thing we lose with this is a [gha problem matcher](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md).

We can add one easily, but I feel like dtolnay/rust-toolchain should do that, but I know the author wants to keep it minimal

**Review by burrbull — APPROVED** [review]

**Comment by burrbull** [comment]
Do we need to add `merge_group`?

**Comment by burrbull** [comment]
I messed up svd2rust and svd PRs.

**Comment by Emilgardis** [comment]
this does not implement merge queue, did you enable it here @burrbull ? Going to do that separately

**Comment by Emilgardis** [comment]
#727 implements merge_group

**Pull request merged**